packages/react/src/components/Panel/index.test.tsx

```diff
@@ -0,0 +1,259 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import Panel, { PanelHeader, PanelContent } from './';
+import axe from '../../axe';
+
+test('should render the default heading level', () => {
+  render(<Panel heading={'Title'}>Content</Panel>);
+
+  expect(screen.getByRole('heading', { name: 'Title' }).tagName).toBe('H2');
+  expect(screen.getByText('Content')).toBeInTheDocument();
+});
+
+test('should render a custom heading level', () => {
+  render(<Panel heading={{ text: 'Title', level: 3 }}>Content</Panel>);
+
+  expect(screen.getByRole('heading', { name: 'Title' }).tagName).toBe('H3');
+  expect(screen.getByText('Content')).toBeInTheDocument();
+});
+
+test('should render the label the panel by the heading by default', () => {
+  render(<Panel heading={'Title'}>Content</Panel>);
+
+  const panel = screen.getByRole('region');
+  const heading = screen.getByRole('heading', { name: 'Title' });
+  const body = screen.getByText('Content');
+
+  const headingId = heading.getAttribute('id');
+  const ariaLabelledBy = panel.getAttribute('aria-labelledby');
+
+  expect(heading.tagName).toBe('H2');
+  expect(ariaLabelledBy).toBe(headingId);
+  expect(body).toBeInTheDocument();
+});
+
+test('should render the label the panel by the heading id', () => {
+  render(
```

Comment:
This is similar to the above, except that it should support the custom id for the heading.

Reply:
I've revised it to make more sense.